wgpu-hal/src/gles/adapter.rs

```diff
@@ -349,6 +393,29 @@ impl crate::Adapter<super::Api> for super::Adapter {
         let zeroes = vec![0u8; super::ZERO_BUFFER_SIZE];
         gl.buffer_data_u8_slice(glow::COPY_READ_BUFFER, &zeroes, glow::STATIC_DRAW);
 
+        // Compile the shader program we use for doing manual clears to work around Mesa fastclear
+        // bug.
+        //
+        // Compile in a closure so we can use `?` to unwrap the errors and handle them all once
+        let (shader_clear_program, shader_clear_program_color_uniform_location) = (|| {
```

Comment:
honestly, I'd rather see `unwrap()` here without any closures. Debugging closures is inconvenient, and handling failures here specifically isn't very actionable for the user anyway